From bc38b9d85bc11632745a040edfd12bfe0954b160 Mon Sep 17 00:00:00 2001
From: Marko Lindqvist <cazfi74@gmail.com>
Date: Fri, 3 May 2024 23:34:48 +0300
Subject: [PATCH 61/61] savegame: Correct loading last turn change time

The value was erroneously rounded down to full seconds,
while it's stored with 1/100 second accuracy in the savegame.

Reported by alain_bkr

See RM #545

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>
---
 server/savegame/savegame2.c | 2 +-
 server/savegame/savegame3.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/server/savegame/savegame2.c b/server/savegame/savegame2.c
index f5bbd75fdf..ad319a2033 100644
--- a/server/savegame/savegame2.c
+++ b/server/savegame/savegame2.c
@@ -1971,7 +1971,7 @@ static void sg_load_game(struct loaddata *loading)
     = !secfile_lookup_bool_default(loading->file, TRUE, "game.save_players");
 
   game.server.turn_change_time
-    = secfile_lookup_int_default(loading->file, 0, "game.last_turn_change_time") / 100;
+    = secfile_lookup_int_default(loading->file, 0, "game.last_turn_change_time") / 100.0;
 }
 
 /* =======================================================================
diff --git a/server/savegame/savegame3.c b/server/savegame/savegame3.c
index 31a2ed11e0..f3d8fbed42 100644
--- a/server/savegame/savegame3.c
+++ b/server/savegame/savegame3.c
@@ -2234,7 +2234,7 @@ static void sg_load_game(struct loaddata *loading)
     = !secfile_lookup_bool_default(loading->file, TRUE, "game.save_players");
 
   game.server.turn_change_time
-    = secfile_lookup_int_default(loading->file, 0, "game.last_turn_change_time") / 100;
+    = secfile_lookup_int_default(loading->file, 0, "game.last_turn_change_time") / 100.0;
 }
 
 /************************************************************************//**
-- 
2.43.0

